model_compression_toolkit/qat/keras/quantizer/ste_rounding/symmetirc_ste.py

```diff
@@ -0,0 +1,145 @@
+    """
+
+    def __init__(self, qunatization_config: NodeWeightsQuantizationConfig):
+        """
+        Initialize a TrainableWeightQuantizer object with parameters to use
+        for the quantization.
+
+        Args:
+            qunatization_config: Number of bits to use for the quantization.
+        """
+        super().__init__(qunatization_config,
+                         qi.QuantizationPart.Weights,
+                         [qi.QuantizationMethod.POWER_OF_TWO, qi.QuantizationMethod.SYMMETRIC])
+        self.power_of_two = qunatization_config.weights_quantization_method == QuantizationMethod.POWER_OF_TWO
+        self.threshold_values = qunatization_config.weights_quantization_params[C.THRESHOLD]
+        self.threshold_shape = np.asarray(self.threshold_values).shape
+        self.np_threshold_values = np.reshape(np.asarray(self.threshold_values),
+                                              [-1]) if self.qunatization_config.weights_channels_axis else float(
+            self.threshold_values)
+
+        if self.qunatization_config.weights_per_channel_threshold and self.qunatization_config.weights_channels_axis not in [
+            -1, len(self.threshold_shape) - 1]:
+            # Tensorflow's fake_quant_with_min_max_vars_per_channel only works on last axis, so
+            # need to move the quantization axis to the last axis
+            self.perm_vec = list(np.arange(len(self.threshold_shape)))
+            self.perm_vec[self.qunatization_config.weights_channels_axis] = len(self.threshold_shape) - 1
+            self.perm_vec[len(self.threshold_shape) - 1] = self.qunatization_config.weights_channels_axis
+        else:
+            self.perm_vec = None
+
+        if self.power_of_two:
+            self.np_threshold_values = np.power(2.0,
+                                                np.ceil(np.log2(np.maximum(self.np_threshold_values, C.MIN_THRESHOLD))))
+        num_bits = self.qunatization_config.weights_n_bits
+        delta = self.np_threshold_values / np.power(2.0, num_bits - int(C.WEIGHTS_SIGNED))
+        min_int = -int(C.WEIGHTS_SIGNED) * (2 ** (num_bits - int(C.WEIGHTS_SIGNED)))
+        max_int = (2 ** (num_bits - int(C.WEIGHTS_SIGNED))) - 1
+        self.min = delta * min_int
+        self.max = delta * max_int
+        self.quantizer_parameters = {}
+
+    def initialize_quantization(self,
+                                tensor_shape: TensorShape,
+                                name: str,
+                                layer: QuantizeWrapper) -> Dict[str, tf.Variable]:
+        """
+        Add min and max variables to layer.
+        Args:
+            tensor_shape: Tensor shape the quantizer quantize.
+            name: Prefix of variables names.
+            layer: Layer to add the variables to. The variables are saved
+            in the layer's scope.
+
+        Returns:
+            Dictionary of new variables.
+        """
+        ptq_threshold_tensor = layer.add_weight(
```

Comment:
maybe instead of all these weights we can only add the "delta" weight. along with the number of bits attribute it should suffice.